x/gov/keeper/proposal.go

```diff
@@ -11,7 +11,11 @@ import (
 )
 
 // SubmitProposal create new proposal given an array of messages
-func (keeper Keeper) SubmitProposal(ctx sdk.Context, messages []sdk.Msg) (v1beta2.Proposal, error) {
+func (keeper Keeper) SubmitProposal(ctx sdk.Context, messages []sdk.Msg, metadata []byte) (v1beta2.Proposal, error) {
+	if metadata != nil && len(metadata) > int(keeper.maxMetadataLen) {
+		return v1beta2.Proposal{}, types.ErrMetadataTooLong.Wrapf("got metadata with length %d", len(metadata))
+	}
```

Comment:
don't we want to add this metadata to the proposal?

Reply:
In the error case? in my experience, if there's `a, err := ...; err !=nil`, people rarely use `a`

I can add it though

Reply:
I mean at the moment we're not adding the `metadata` to the actual `Proposal` struct. We need to add the argument to the `NewProposal` function.

Reply:
Oh yeah you're totally right. I added.